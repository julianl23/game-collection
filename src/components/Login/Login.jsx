import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import TextField from "../TextField";

const Login = () => {
  const LoginSection = styled.section`
    grid-column: 2 / span 2;
  `;

  const Heading = styled.h2`
    font-size: ${props => props.theme.fontSizeXLarge};
    font-weight: 500;
    text-align: center;
  `;

  return (
    <LoginSection>
      <Heading>Log In</Heading>
      <form action="/" method="POST">
        <TextField id="email" labelText="Email" type="email" />
        <TextField id="password" labelText="Password" type="password" />
      </form>
    </LoginSection>
  );
};

Login.propTypes = {
  theme: PropTypes.object,
};

Login.defaultProps = {
  theme: {},
};

export default Login;
