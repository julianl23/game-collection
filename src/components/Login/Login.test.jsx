import React from "react";
import TestRenderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import Login from "./Login";

describe("Login", () => {
  it("renders correctly", () => {
    const instance = TestRenderer.create(
      <ThemeProvider theme={{}}>
        <Login />
      </ThemeProvider>
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
