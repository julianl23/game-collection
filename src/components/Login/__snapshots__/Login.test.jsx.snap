// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<section>
  It me, the login
</section>
`;
